libs/datamodel/core/src/transform/ast_to_dml/lift.rs

```diff
@@ -194,24 +193,74 @@ impl<'a> LiftAstToDml<'a> {
     ) -> Result<(dml::FieldType, Vec<ast::Directive>), DatamodelError> {
         let type_name = &ast_field.field_type.name;
 
+        let mut supports_native_types = false;
+        let mut datasource_name = "";
+        match self.source {
+            Some(source) => {
+                supports_native_types = source.has_preview_feature("nativeTypes");
+                datasource_name = source.name.as_str();
+            }
+            _ => {}
+        };
+
         if let Ok(scalar_type) = ScalarType::from_str(type_name) {
-            if USE_CONNECTORS_FOR_CUSTOM_TYPES {
-                let pg_connector = SqlDatamodelConnectors::postgres();
-                let pg_type_specification = ast_field.directives.iter().find(|dir| dir.name.name.starts_with("pg.")); // we use find because there should be at max 1.
-                let name = pg_type_specification.map(|dir| dir.name.name.trim_start_matches("pg."));
-                let args = pg_type_specification
-                    .map(|dir| {
-                        let args = dir
-                            .arguments
-                            .iter()
-                            .map(|arg| ValueValidator::new(&arg.value).as_int().unwrap() as u32)
-                            .collect();
-                        args
-                    })
-                    .unwrap_or(vec![]);
-
-                if let Some(x) = name.and_then(|ts| pg_connector.parse_native_type(&ts, args)) {
-                    let field_type = dml::FieldType::NativeType(scalar_type, x);
+            if supports_native_types {
+                let connector_string = match self.source {
+                    Some(source) => &source.active_provider,
+                    None => "",
+                };
+                let mut connectors: HashMap<&str, Box<dyn Connector>> = HashMap::new();
+                connectors.insert("postgresql", Box::new(SqlDatamodelConnectors::postgres()));
+                connectors.insert("mysql", Box::new(SqlDatamodelConnectors::mysql()));
+                connectors.insert("sqlite", Box::new(SqlDatamodelConnectors::sqlite()));
+                connectors.insert("mssql", Box::new(SqlDatamodelConnectors::mssql()));
+
+                let mut prefix = String::with_capacity(datasource_name.len() + 1);
+                prefix.push_str(datasource_name);
+                prefix.push_str(".");
+
+                let mut type_specifications = ast_field
+                    .directives
+                    .iter()
+                    .filter(|dir| dir.name.name.starts_with(&prefix));
+
+                let type_specification = type_specifications.next();
+
+                if type_specifications.count() > 1 {
+                    return Err(DatamodelError::new_duplicate_directive_error(
+                        &prefix,
+                        type_specification.unwrap().span,
+                    ));
+                }
```

Comment:
This logic might not do what you think. Assume there were two type specifications in the schema. The call to next will remove one item from the iterator. Hence the count invocation will return `1` instead of the original `2` you might have expected.